doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -266,6 +285,36 @@ More on sorts can be found in Section :ref:`sorts`.
 Binders
 -------
 
+.. insertgram open_binders exclam_opt
+
+.. productionlist:: coq
+   open_binders : `name_list` : `term`
+                : `binders`
+   name_list : `name_list` `name`
+             : `name`
+   name : _
+        : `ident`
+   binders : `binders` `binder`
+           : `binder`
+   binder : `name`
+          : ( `name_list` : `term` )
+          : ( `name` `colon_term_opt` := `term` )
+          : { `name` }
+          : { `name_list` `colon_term_opt` }
+          : `( `typeclass_constraint_list_comma` )
+          : `{ `typeclass_constraint_list_comma` }
+          : ' `pattern0`
+          : ( `name` : `term` | `term` )
+   typeclass_constraint_list_comma : `typeclass_constraint_list_comma` , `typeclass_constraint`
```

Comment:
I made this `typeclass_constraints_comma` and removed "_list" from a number of other names.  I think it's helpful to include the name of the separator in the nonterminal so the user doesn't need to jump back and forth so much in the productionlist form of the grammar.  Also, BTW, most of the symbols ending in "_list", "_opt", etc. are automatically generated by doc_gram--including the separator helps generate an understandable unique nonterminal name (better than adding arbitrary digits to the end of the name for uniqueness).

If we switch to using prodn, then a lot of these lists will be inlined and we won't have to invent a nonterminal name.

Reply:
OK, why not to include the name of the separator in the nonterminal then. What worries me is to have a consistent style all over the manual, thus later changing all grammars in all chapters along the same pattern.

If we switch to prodn, this would be another way to address consistency (afaiu).

Reply:
> What worries me is to have a consistent style all over the manual, thus later changing all grammars in all chapters along the same pattern.

Consistency is my goal.